### PR TITLE
add Image model

### PR DESCRIPTION
**Pull request by MixailE**

### CODE REVIEW

File: src/models/Image.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      image.generating = true
+    }
+    await image.save()
+
+    try {
+      const readableStream = await generateAndDownloadImage(username, 3)
+      const { cid } = await uploadToIpfs(readableStream)
+      image.cid = cid
+    } catch (e) {
+      console.error(e)
```
Comment (T-Damer) [inline src/models/Image.ts:35]: I think we should report about pfp errors in sentry as well

File: src/models/Image.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      throw e
+    } finally {
+      image.generating = false
+      await image.save()
+    }
+  }
+
+  if (image.generating) {
+    await new Promise((res) => setTimeout(res, 5000))
```
Comment (T-Damer) [inline src/models/Image.ts:44]: Let's put this into `sleep.ts` helper or `wait.ts`

File: src/models/Image.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    return findOrCreateImage(username)
```
Comment (T-Damer) [inline src/models/Image.ts:45]: Also, it looks like it can get into infinite loop

**Review by T-Damer — APPROVED** [review]

File: src/models/Image.ts (hunk 1 above)
Comment (MixailE) [reply src/models/Image.ts:40]: We should create new task for that

**Comment by MixailE** [comment]
[Fix PFPs prompt to generate better images](https://trello.com/c/4OrNT0p8/4420-fix-pfps-prompt-to-generate-better-images)

File: src/controllers/profile.ts [hunk 4]
```diff
@@ -1,20 +1,44 @@
...
   @Post('/')
   async picGenerator(
     @Body({ required: true })
-    { address }: Address
+    { address }: Address,
+    @Ctx() ctx: Context
   ) {
     const profilePicture = await findOrCreateProfilePicture(address)
+
+    if (!profilePicture) ctx.throw(notFound())
```
[the remark this answers is not in the document]
Comment (T-Damer) [reply src/controllers/profile.ts:20]: Agree

**Pull request closed without merging**